sdk/storage/inc/datalake/datalake_options.hpp

```diff
@@ -217,65 +228,41 @@ namespace Azure { namespace Storage { namespace DataLake {
      *        Properties" will not return this property unless it is explicitly set on that file
      *        again.
      */
-    std::string ContentMD5;
+    Azure::Core::Nullable<std::string> ContentMD5;
 
     /**
      * @brief The lease ID must be specified if there is an active lease.
      */
-    std::string LeaseId;
-
-    /**
-     * @brief Sets the path's cache control. If specified, this property is stored with the
-     *        path and returned with a read request.
-     */
-    std::string CacheControl;
-
-    /**
-     * @brief Sets the path's content type. If specified, this property is stored
-     *        with the path and returned with a read request.
-     */
-    std::string ContentType;
-
-    /**
-     * @brief Sets the path's Content-Disposition header.
-     */
-    std::string ContentDisposition;
-
-    /**
-     * @brief Sets the path's content encoding. If specified, this property is stored with
-     *        the path and returned with a read request.
-     */
-    std::string ContentEncoding;
+    Azure::Core::Nullable<std::string> LeaseId;
 
     /**
-     * @brief Set the path's content language. If specified, this property is stored with
-     *        the path and returned with a read request.
+     * @brief Specify the http headers for this path.
      */
-    std::string ContentLanguage;
+    Azure::Storage::DataLake::DataLakeHttpHeaders HttpHeaders;
 
     /**
      * @brief Specify an ETag value to operate only on path with a matching value.
      */
-    std::string IfMatch;
+    Azure::Core::Nullable<std::string> IfMatch;
 
     /**
      * @brief Specify an ETag value to operate only on path without a matching value.
      */
-    std::string IfNoneMatch;
+    Azure::Core::Nullable<std::string> IfNoneMatch;
 
     /**
      * @brief A date and time value. Specify this header to perform the
      *        operation only if the resource has been modified since the
      (        specified date and time.
     */
-    std::string IfModifiedSince;
+    Azure::Core::Nullable<std::string> IfModifiedSince;
 
     /**
      * @brief A date and time value. Specify this header to perform the
      *        operation only if the resource has not been modified since
      *        the specified date and time.
      */
-    std::string IfUnmodifiedSince;
+    Azure::Core::Nullable<std::string> IfUnmodifiedSince;
```

Comment:
This is different than how done for blobs since we added structures to blobs. Maybe we should be consistent across client libraries? Thoughts?

Reply:
Yes, waiting for that PR to be merged to abstract access conditions.